app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts

```diff
@@ -117,55 +145,129 @@ async function walletCreateSessionHandler(
+        MultichainNetwork.Solana,
+      );
+      promptToCreateSolanaAccount = supportedSolanaAccounts.length === 0;
+    }
+
     const { supportedScopes: supportedRequiredScopes } = bucketScopes(
       requiredScopesWithSupportedMethodsAndNotifications,
       {
-        isChainIdSupported: networkClientExistsForChainId,
-        isChainIdSupportable: () => false, // intended for future usage with eip3085 scopedProperties
+        isEvmChainIdSupported: networkClientExistsForChainId,
+        isEvmChainIdSupportable: () => false, // intended for future usage with eip3085 scopedProperties
+        getNonEvmSupportedMethods: hooks.getNonEvmSupportedMethods,
+        isNonEvmScopeSupported: hooks.isNonEvmScopeSupported,
       },
     );
 
     const { supportedScopes: supportedOptionalScopes } = bucketScopes(
       optionalScopesWithSupportedMethodsAndNotifications,
       {
-        isChainIdSupported: networkClientExistsForChainId,
-        isChainIdSupportable: () => false, // intended for future usage with eip3085 scopedProperties
+        isEvmChainIdSupported: networkClientExistsForChainId,
+        isEvmChainIdSupportable: () => false, // intended for future usage with eip3085 scopedProperties
+        getNonEvmSupportedMethods: hooks.getNonEvmSupportedMethods,
+        isNonEvmScopeSupported: hooks.isNonEvmScopeSupported,
       },
     );
 
+    const allRequestedAccountAddresses = getAllAccountsFromScopesObjects([
+      supportedRequiredScopes,
+      supportedOptionalScopes,
+    ]);
+
+    const allSupportedRequestedCaipChainIds = getAllScopesFromScopesObjects([
+      supportedRequiredScopes,
+      supportedOptionalScopes,
+    ]);
+
     // Fetch EVM accounts from native wallet keyring
-    // These addresses are lowercased already
     const existingEvmAddresses = hooks
       .listAccounts()
       .map((account) => account.address);
-    const supportedEthAccounts = getEthAccounts({
-      requiredScopes: supportedRequiredScopes,
-      optionalScopes: supportedOptionalScopes,
-    })
-      .map((address) => address.toLowerCase() as Hex)
-      .filter((address) => existingEvmAddresses.includes(address));
+
+    const supportedRequestedAccountAddresses =
+      allRequestedAccountAddresses.filter((accountAddress: CaipAccountId) => {
+        const {
+          address,
+          chain: { namespace },
+          chainId: caipChainId,
+        } = parseCaipAccountId(accountAddress);
+        if (namespace === KnownCaipNamespace.Eip155) {
+          return existingEvmAddresses.some((existingEvmAddress) => {
+            return isEqualCaseInsensitive(address, existingEvmAddress);
+          });
+        }
+
+        const getNonEvmAccountAddressesForChainId =
+          hooks.getNonEvmAccountAddresses(caipChainId);
+        return getNonEvmAccountAddressesForChainId.some(
+          (existingCaipAddress) => {
+            return isEqualCaseInsensitive(accountAddress, existingCaipAddress);
```

Comment:
Checking what are we doing on `isEqualCaseInsensitive`, because a Solana address can be a different one with a single letter in lowercase. If we are lowercasing and the checking equality, then we might want to look at other places where this is also used.

Reply:
Ok I've resolved atleast this case in these commits: 
- https://github.com/MetaMask/metamask-extension/pull/29887/commits/97660e51b541c07dd8598467dbe1e7393bf49edc
- https://github.com/MetaMask/metamask-extension/pull/29887/commits/29af17be43b8db6d68e5f18ff7b5f7e517c999f4

we need to loop back and make sure there aren't other cases where this is an issue though